File: src/main/java/fi/finlit/edith/ui/services/hibernate/PersonDaoImpl.java
package fi.finlit.edith.ui.services.hibernate;

import static fi.finlit.edith.sql.domain.QPerson.person;

import java.util.Collection;

import fi.finlit.edith.sql.domain.Person;
import fi.finlit.edith.ui.services.PersonDao;

public class PersonDaoImpl extends AbstractDao<Person> implements PersonDao {
    @Override
    public Collection<Person> findByStartOfFirstAndLastName(String partial, int limit) {
        return query()
                .from(person)
                .where(person.normalizedForm.first.startsWithIgnoreCase(partial).or(
                        person.normalizedForm.last.startsWithIgnoreCase(partial))).limit(limit)
                .list(person);
    }

    @Override
    public void remove(Long personId) {
        // FIXME: Hibernatify!
        Person entity = getById(personId);
        if (entity != null) {
            getSession().delete(entity);
        }
    }

    @Override
    public void save(Person person) {
        getSession().save(person);
    }

    @Override
    public void remove(Person person) {
        getSession().delete(person);
    }

    @Override
    public Collection<Person> getAll() {
        return query().from(person).list(person);
    }

    @Override
    public Person getById(Long id) {
        return query().from(person).where(person.id.eq(id)).uniqueResult(person);
    }

}

File: src/main/scala/fi/finlit/edith/ui/services/hibernate/PersonDaoImpl.scala
package fi.finlit.edith.ui.services.hibernate

import fi.finlit.edith.sql.domain.QPerson.person
import java.util.Collection
import fi.finlit.edith.sql.domain.Person
import fi.finlit.edith.ui.services.PersonDao
//remove if not needed
import scala.collection.JavaConversions._

class PersonDaoImpl extends AbstractDao[Person] with PersonDao {

  override def findByStartOfFirstAndLastName(partial: String, limit: Int): Collection[Person] = {
    query.from(person).where(person.normalized.first.startsWithIgnoreCase(partial).or(person.normalized.last.startsWithIgnoreCase(partial))).limit(limit).list(person)
  }

  override def remove(personId: java.lang.Long) {
    var entity = getById(personId)
    if (entity != null) {
      getSession.delete(entity)
    }
  }

  override def save(person: Person) {
    getSession.save(person)
  }

  override def getById(id: java.lang.Long): Person = {
    query.from(person).where(person.id eq id).uniqueResult(person)
  }
}
